### ISSUE TITLE
Mangadex is not updating the list

### ISSUE THREAD

**Issue by hjp78** [dazedcat19/FMD#865]
- [ ] I'm using the latest version.
- [ ] All website modules are up to date.
- [ ] This is not temporary server issue.

**System information**
Windows 11

**Describe the bug**
Since 2/3 days ago, Mangadex is not updating the manga list.

**Comment by MrTyton** [comment]
+1. Can access mangadex through browser, but FMD has stopped being able to access list or download anything

**Comment by hjp78** [comment]
> +1. Can access mangadex through browser, but FMD has stopped being able to access list or download anything

In my case, i still can download from my list (I'm not using a VPN)

**Comment by muto4013** [comment]
Same here, in log i have the following:

EXCEPTION: LUA>DoGetNameAndLink("MangaDex.lua")> LuaCallFunction("GetNameAndLink") LUA_ERRRUN: C:\Program Files\fmd_2\lua\modules\MangaDex.lua:110: attempt to perform arithmetic on a nil value

Hope this helps

**Comment by fyram9** [comment]
I took a stab at this despite not knowing any lua, and after a few hours of trial and error, it seems this is because some sort of rate limiter on MangaDex side (that is undocumented on their [site](https://api.mangadex.org/docs.html#section/Rate-limits), probably from their DDOS protection or something).

So what I did is add `Delay()` on MangaDex.lua at line 89 (right after `while total > offset do`).
![image](https://user-images.githubusercontent.com/97740853/153694135-49a79b0e-b6e9-43fb-a46a-11944a5e9271.png)

Then, I set the delay for MangaDex in the option to 5 seconds (or maybe higher if you want to be safer).
![image](https://user-images.githubusercontent.com/97740853/153694154-2151cf33-2237-4b6b-9501-f7953c63fc38.png)

This works for me and FMD now updating correctly.
